kolibri/plugins/coach/assets/src/views/lessons/LessonSummaryPage/UpdatedResourceSelection.vue

```diff
@@ -125,12 +125,20 @@
     },
     methods: {
       contentLink(content) {
-        const { name, params, query } = this.$route;
+        const { params, query } = this.$route;
         if (!content.is_leaf) {
           return this.topicsLink(content.id);
         }
-        // Just return the current route; router-link will handle the no-op from here
-        return { name, params, query };
+        return {
+          name: PageNames.LESSON_PREVIEW_RESOURCE,
+          params: {
+            ...params,
+          },
+          query: {
+            contentId: content.id,
+            ...query,
```

Comment:
One thing to note here is that if `query.contentId` is already set in the `query` object you take from `params` above here, then you'll be overwriting it w/ what is already set in `query`.

When values are destructured into an object like this, values are applied in the order they're listed.

For example:

```javascript
const x = { a: 1, b: 2, c: 3};
// We're destructuring x into y after we assign the value to `a`
const y = { a: 50, ...x };    
console.log("y.a: ", y.a);  // y.a: 1

// Now x is destructured into z before we assign `a`
const z = { ...x, a: 40 };
// So the value assigned to `a` is reflected in the `z` object
console.log("z.a: ", z.a)  // z.a: 40
```